engines/ags/dialogs.cpp

```diff
@@ -75,6 +76,9 @@ AGSOptionsWidget::AGSOptionsWidget(GuiObject *boss, const Common::String &name,
 		_langPopUp->appendEntry(traFileName, i++);
 	}
 
+	// Override game save management
+	_overrideSavesCheckbox = new GUI::CheckboxWidget(widgetsBoss(), _dialogLayout + ".savesOvr", _("Always use ScummVM save management"), _("Always enable ScummVM's save management and autosaves, even for games that are not fully compatible (NOT RECOMMENDED)"));
```

Comment:
What does "not fully compatible" mean in this context?

Reply:
Yeah that tooltip isn't too great, but I couldn't come up with something better... basically it means: "even for games that should manage saves (and autosaves) autonomously" (because ScummVM's features are disabled in the detection entry, that is)

Reply:
How about

```suggestion
	_overrideSavesCheckbox = new GUI::CheckboxWidget(widgetsBoss(), _dialogLayout + ".savesOvr", _("Always use ScummVM save management"), _("Do not allow games to use their built-in manual save and autosave management, and instead use ScummVM manual save and autosave management (NOT RECOMMENDED)"));
```

Also, why is it not recommended?

Reply:
This sounds like the two options are mutually exclusive, but in reality it's:
- saves managed by the game, or
- saves managed by the game + scummvm
I have another proposal: "Never disable ScummVM save management, even for games that should manually handle saves and autosaves"
I could probably drop the "NOT RECOMMENDED", seems a bit too strong

Reply:
> I have another proposal: "Never disable ScummVM save management, even for games that should manually handle saves and autosaves"

Could we just flip the option: "Disable ScummVM save management"?

Part of the confusion may be because this PR is addressing two issues:

1. Games like _The Shivah_ have in-game autosaves and ScummVM autosaves either duplicating effort or interfering with each other
2. Games like _Dustbowl_ or _Whispers of a Machine_ intentionally prevent manual saves, but ScummVM currently allows them

It seems to me like we might actually want two options:

1. Disable ScummVM auto-saving (prevents conflict with games that do their own autosaving)
2. Disable ScummVM manual saves (useful for games that allow permadeath, iron man, etc)

Reply:
For what it's worth, here's a summary of how various games work in the original interpreter:

Game | In-Game Manual Saves | In-Game Auto-Saves
--- | --- | --- 
_Steamed Hams_ | ❌ | ❌
Typical AGS game? | ✅ | ❌
_Whispers of a Machine_ | ❌ | ✅
_The Blackwell Legacy_ | ✅ | ✅ 

And here's how they currently work in ScummVM:

Game | In-Game Manual Saves | ScummVM Manual Saves | In-Game Auto-Saves | ScummVM Auto-Saves | Notes
--- | --- | --- | --- | --- | ---
_Steamed Hams_ | ❌ | ✅ | ❌ | ✅ | Extra save options are nice
Typical AGS game? | ✅ | ✅ | ❌ | ✅ | Manual saves either way are fine, extra autosaves are nice
_Dustbowl_ | ❌ | ✅ | ✅ | ✅ | Manual saves are against the spirit of the game, but may be desired. In-game and ScummVM autosaves interfere
_The Blackwell Legacy_ | ✅ | ✅ | ✅ | ✅ | In-game and ScummVM autosaves interfere

Note that for some games it's a good thing to have ScummVM allowing manual saves when it wasn't offered and for some games it's not, and the same with autosaves.

Reply:
Not sure about this... I wanted to keep things simple from a user point of view:
- If you leave the checkbox unchecked, ScummVM uses the settings setup internally (which _may_ disable the GMM saves or autosaves)
- If you check the box, all the usual features (autosave and GMM saving) are always enabled, but you take the chance that there may be inconsistencies

If we decide to use two "disable ..." checkboxes, besides the fact that the entire thing works in reverse (that means that the boxes must be automatically ticked on addition of e.g. Dustbowl, or the user wouldn't bother changing the option)
it could bring additional confusion:
Let's say that the uses ticks "disable scummvm manual saves" for a generic game:
- if we silently ignore the option, it seems the checkbox has no effect in general,
- if we enforce the option, it just disables a perfectly working functionality to no gain

Also, these settings assume that the user knows well how the games and scummvm handle saves, which imho shouldn't be taken for granted...
I'd rather wait for more feedback before going one way or the other

Reply:
The handling of autosaves isn't something that should be decided by the users. This should be handled by the engine itself, enforcing either the ScummVM semantics or the game ones. IMHO, games that handle autosaving on their own should have an option in the detection entry, which will always override and disable the in-built ScummVM autosave functionality.

Regarding the regular save functionality: I believe it's superfluous to have a checkbox for that. Since the save/load menu is a global functionality, there is no point to disable it with a checkbox: users may access it if they wish, which is cheating in some games, but it's their choice. If they want to avoid cheating, they don't need an extra checkbox to ensure that they won't cheat :)

Reply:
Disabling GMM saving is not only to prevent cheating but also to prevent using the functionality in games where it doesn't work properly (such as Strangeland), so the player has to use the game menu instead of scummvm's.

I agree with your first point regarding autosaves, however I decided to add the checkbox basically to keep the current functionality... ticking it disables this PR's effects :)
All things considered i think that a single checkbox is the cleanest implementation.

Reply:
> I agree with your first point regarding autosaves, however I decided to add the checkbox basically to keep the current functionality...

Are there situations where we _would_ want the current functionality? I'm thinking for games like _The Blackwell Legacy_ and _The Shivah_ where there are double autosaves, I don't think the current behavior would ever be desired.

Reply:
Regardless of the checkbox, this PR is temporarily on hold until PR #5496 is merged (since it uses systemwide functions that are directly touched by that PR)